### PR TITLE
Release resources in ChildFirstClassLoaderTest correctly

### LINKED ISSUE #19173: com.hazelcast.jet.impl.deployment.ChildFirstClassLoaderTest.com.hazelcast.jet.impl.deployment.ChildFirstClassLoaderTest fails on Win Oracke JDK 8
mater, revafbee062bba410a246fd2b865360fe789824867a
Failed on Windows, Oracle JDK 8
http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-OracleJDK8/302/testReport/

Stacktrace:
```
java.nio.file.FileSystemException: 
C:\Users\Administrator\AppData\Local\Temp\resources_3201490181683974308.jar: The process cannot access the file because it is being used by another process.

	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:86)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:269)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at com.hazelcast.jet.impl.deployment.ChildFirstClassLoaderTest.afterClass(ChildFirstClassLoaderTest.java:54)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunAfters.invokeMethod(RunAfters.java:46)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418
```

Standard output:
```
15:23:28,478  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
15:23:28,480  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
15:23:28,480  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
15:23:28,480  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210725 - afbee06) starting at [127.0.0.1]:5701
15:23:28,480  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Cluster name: dev
15:23:28,489  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
15:23:28,496  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
15:23:28,499  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
15:23:28,500  INFO || - [Diagnostics] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
15:23:28,500  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5701 is STARTING
15:23:28,500  INFO || - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 9ea4ec91-be9b-4f11-90b9-b789d02cbe64 this
]

15:23:28,500  INFO || - [JetExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Jet is enabled
15:23:28,501  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] [127.0.0.1]:5701 is STARTED
15:23:28,506 DEBUG || - [JobCoordinationService] hz.dazzling_rubin.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
15:23:28,506  INFO || - [ClientInvocationService] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] Running with 2 response threads, dynamic=true
15:23:28,507  INFO || - [LifecycleService] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210725 - afbee06) is STARTING
15:23:28,507  INFO || - [LifecycleService] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210725 - afbee06) is STARTED
15:23:28,508  INFO || - [ClientConnectionManager] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] Trying to connect to cluster: dev
15:23:28,508  INFO || - [ClientConnectionManager] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] Trying to connect to [127.0.0.1]:5701
15:23:28,508  INFO || - [HazelcastClient] Time-limited test - Created connection to endpoint: [127.0.0.1]:5701, connection: MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=null, lastReadTime=never, lastWriteTime=never, closedTime=never, connected server version=null}}
15:23:28,508 DEBUG || - [AuthenticationMessageTask] hz.dazzling_rubin.generic-operation.thread-1 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Processing authentication with clientUuid 80d729ae-8756-42aa-ad17-ac1c8199941a
15:23:28,508  INFO || - [AuthenticationMessageTask] hz.dazzling_rubin.generic-operation.thread-1 - [127.0.0.1]:5701 [dev] [5.0-SNAPSHOT] Received auth from MockedNodeConnection{ remoteAddress = [127.0.0.1]:40001, localAddress = [127.0.0.1]:5701, connectionId = 1}, successfully authenticated, clientUuid: 80d729ae-8756-42aa-ad17-ac1c8199941a, client version: 5.0-SNAPSHOT
15:23:28,508  INFO || - [LifecycleService] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210725 - afbee06) is CLIENT_CONNECTED
15:23:28,508  INFO || - [ClientConnectionManager] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] Authenticated with server [127.0.0.1]:5701:9ea4ec91-be9b-4f11-90b9-b789d02cbe64, server version: 5.0-SNAPSHOT, local address: /127.0.0.1:40001
15:23:28,509  INFO || - [Diagnostics] Time-limited test - hz.client_4339 [dev] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
15:23:28,509 TRACE || - [ClientListenerService] hz.client_4339.internal-1 - hz.client_4339 [dev] [5.0-SNAPSHOT] Register attempt of ClusterViewListenerHandler to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-07-25 15:23:28.508, lastWriteTime=2021-07-25 15:23:28.508, closedTime=never, connected server version=5.0-SNAPSHOT}}
15:23:28,509 TRACE || - [ClientListenerService] ForkJoinPool.commonPool-worker-470 - hz.client_4339 [dev] [5.0-SNAPSHOT] Registered ClusterViewListenerHandler to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-07-25 15:23:28.509, lastWriteTime=2021-07-25 15:23:28.509, closedTime=never, connected server version=5.0-SNAPSHOT}}
15:23:28,509  INFO || - [ClientClusterService] hz.client_4339.event-70739 - hz.client_4339 [dev] [5.0-SNAPSHOT] 

Members [1] {
	Member [127.0.0.1]:5701 - 9ea4ec91-be9b-4f11-90b9-b789d02cbe64
}

15:23:28,510 TRACE || - [ClientListenerService] hz.client_4339.eventRegistration- - hz.client_4339 [dev] [5.0-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@1c9ce4a7, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@12100a8e} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-07-25 15:23:28.509, lastWriteTime=2021-07-25 15:23:28.509, closedTime=never, connected server version=5.0-SNAPSHOT}}
15:23:28,511 TRACE || - [ClientListenerService] hz.client_4339.eventRegistration- - hz.client_4339 [dev] [5.0-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@1c9ce4a7, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@12100a8e} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-07-25 15:23:28.511, lastWriteTime=2021-07-25 15:23:28.510, closedTime=never, connected server version=5.0-SNAPSHOT}}
15:23:28,511  INFO || - [ClientStatisticsService] Time-limited test - Client statistics is enabled with period 5 seconds.
Started Running Test: testDeployment_whenAttachNestedDirectory_thenFilesAvailableOnMembers```

### PR DESCRIPTION
**Pull request by frant-hartm**
Fixes #19173

### CODE REVIEW

**Review by ramizdundar — APPROVED** [review]
LGTM

File: hazelcast/src/test/java/com/hazelcast/jet/impl/deployment/ChildFirstClassLoaderTest.java [hunk 1]
```diff
@@ -187,12 +196,14 @@ public void canLoadClassFromIntermediate() throws Exception {
...
     public void canLoadResourceFromIntermediate() throws Exception {
-        URLClassLoader intermediateCl = new URLClassLoader(new URL[]{jarUrl}, ClassLoader.getSystemClassLoader());
+        intermediateCl = new URLClassLoader(new URL[]{jarUrl}, ClassLoader.getSystemClassLoader());
+
         cl = new ChildFirstClassLoader(new URL[]{emptyJarUrl},
                 intermediateCl);
 
         String content = readResource("childfirstclassloader/resource_jar.txt");
         assertThat(content).isEqualTo("resource in jar");
+
```
Comment (ufukyilmaz) [inline hazelcast/src/test/java/com/hazelcast/jet/impl/deployment/ChildFirstClassLoaderTest.java:206]: nit 😄  
```suggestion
```

**Comment by ufukyilmaz** [comment]
run-lab-run

**Review by ufukyilmaz — APPROVED** [review]
It looks good to me.

**Comment by frant-hartm** [comment]
run-lab-run

**Pull request merged**